### ISSUE TITLE
Loss function differs from that in the paper

### ISSUE THREAD

**Issue by JerryLife** [yongqyu/STRNN#6]
In train_torch.py line 98-104, the loss function is implemented as 
```python
    def loss(self, user, td_upper, td_lower, ld_upper, ld_lower, loc, dst, hx):
        h_tq = self.forward(td_upper, td_lower, ld_upper, ld_lower, loc, hx)
        p_u = self.permanet_weight(user)
        q_v = self.location_weight(dst)
        output = torch.mm(q_v, (h_tq + torch.t(p_u)))

        return torch.log(1+torch.exp(torch.neg(output)))
```
However, in the paper, the loss function is as following
![image](https://user-images.githubusercontent.com/16184146/54264854-03fcc000-45af-11e9-87c7-1606e2ba1036.png)
I thought _output_ in the codes refers to _Ou,v,t_ but where is _Ou,v,t'_ in codes? Meanwhile, it seems the regulation part is not considered as well.

**Comment by yongqyu** [comment]
I do not remember exactly the code, but probably no effect of negative-sampling, or the amount of computation would have removed it.
The code for regularization is shown below.
https://github.com/yongqyu/STRNN/blob/e1071448ff207f40fcb2e9f71628babda98e27fd/train_torch.py#L190

I'm sorry I can not give you a helpful answer.
